### PR TITLE
CLM-29805 use NxScrollRender in NxTransferListHalf

### PR DESCRIPTION
**Pull request by obarra-dev**
Jira ticket(s) link: https://sonatype.atlassian.net/browse/CLM-29805
Jenkins build: https://jenkins.ci.sonatype.dev/job/uxui/job/sonatype-react-shared-components/job/CLM-29805-using-NxScrollRender-in-NxTransferListHalf/lastBuild

Note
I tested this improvement on gallery, it seems to work quite well I did not notice the the pages hangs
image

### CODE REVIEW

File: gallery/src/components/NxTransferListHalf/NxTransferListHalfPage.tsx [hunk 1]
```diff
@@ -276,6 +278,13 @@ const NxTransferListPage = () =>
                         liveExample={NxTransferListHalfOrderingWithDisableTransferExample}>
       Demonstrates an ordering <NxCode>NxTransferListHalf</NxCode> with transfer feature disabled.
     </GalleryExampleTile>
+    <GalleryExampleTile title="Long List Example"
+                        id="nx-transfer-list-half-long-list-example"
+                        codeExamples={nxTransferListHalfLongListExample}
+                        liveExample={NxTransferListHalfLongListExample}>
+      Demonstrates an <NxCode>NxTransferListHalf</NxCode> with a long list. Note that even using a long list,
+      rendering and filtering are not affected.
```
Comment (rpokorny) [inline gallery/src/components/NxTransferListHalf/NxTransferListHalfPage.tsx:286]: Maybe should reword slightly to make it clear that you mean affected in a performance sense

**Comment by rpokorny** [comment]
Looks good so far.  Were you still getting test failures?

File: gallery/src/components/NxTransferListHalf/NxTransferListHalfPage.tsx (hunk 1 above)
Comment (obarra-dev) [reply gallery/src/components/NxTransferListHalf/NxTransferListHalfPage.tsx]: > Maybe should reword slightly to make it clear that you mean affected in a performance sense

done at 79dd73fc26f2dd718a4556536c44587e7d0e0825

File: gallery/src/components/NxTransferListHalf/NxTransferListHalfPage.tsx [hunk 2]
```diff
@@ -276,6 +278,14 @@ const NxTransferListPage = () =>
                         liveExample={NxTransferListHalfOrderingWithDisableTransferExample}>
       Demonstrates an ordering <NxCode>NxTransferListHalf</NxCode> with transfer feature disabled.
     </GalleryExampleTile>
+    <GalleryExampleTile title="Long List Example"
+                        id="nx-transfer-list-half-long-list-example"
+                        codeExamples={nxTransferListHalfLongListExample}
+                        liveExample={NxTransferListHalfLongListExample}>
+      Demonstrates an <NxCode>NxTransferListHalf</NxCode> with a long list. This component suffers from performance
+      issues when the list is very long. To address this, it has been updated to use NxScrollRender to render
```
Comment (rpokorny) [inline gallery/src/components/NxTransferListHalf/NxTransferListHalfPage.tsx:286]:
```suggestion
      issues when the list is very long. To address this, it has been updated to use <NxCode>NxScrollRender</NxCode> to render
```

File: gallery/visualtests/nxTransferList.js [hunk 3]
```diff
@@ -47,8 +49,14 @@ describe('NxTransferList', function() {
   });
 
   it('handles overflowing content with a tooltip on items initially scrolled out of view', async function() {
-    const [list, lastItem] = await waitAndGetElements(simpleListSelector, lastItemSelector);
+    const [list] = await waitAndGetElements(simpleListSelector);
+    await scrollIntoView(list);
+
+    const [listHalf] = await waitAndGetElements(listSelector);
+    await listHalf.evaluate(e => e.scrollBy(0, 10000));
+    await wait(500);
```
Comment (rpokorny) [inline gallery/visualtests/nxTransferList.js:57]: This `wait` is probably unnecessary; doesn't `waitAndGetElements` already give the element some time to appear?

File: lib/src/components/NxSearchTransferList/__tests__/NxSearchTransferList.test.tsx [hunk 4]
```diff
@@ -28,6 +28,37 @@ describe('NxSearchTransferList', function() {
       quickRender = rtlRender(NxSearchTransferList as ComponentType<Props<string>>, minimalProps),
       renderEl = rtlRenderElement(NxSearchTransferList as ComponentType<Props<string>>, minimalProps);
 
+  beforeEach(function() {
```
Comment (rpokorny) [inline lib/src/components/NxSearchTransferList/__tests__/NxSearchTransferList.test.tsx:31]: This probably deserves a brief code comment describing the behaviors it mocks

Comment (rpokorny) [inline lib/src/components/NxSearchTransferList/__tests__/NxSearchTransferList.test.tsx:31]: Oh right, I forgot this was copied into several different test files.  Please move it to a utility file in `lib/src/__testutils__`

File: gallery/visualtests/nxTransferList.js (hunk 3 above)
Comment (obarra-dev) [reply gallery/visualtests/nxTransferList.js:57]: > This `wait` is probably unnecessary; doesn't `waitAndGetElements` already give the element some time to appear?

It is needed, If I remove the test fails

File: lib/src/components/NxSearchTransferList/__tests__/NxSearchTransferList.test.tsx (hunk 4 above)
Comment (obarra-dev) [reply lib/src/components/NxSearchTransferList/__tests__/NxSearchTransferList.test.tsx:32]: > Oh right, I forgot this was copied into several different test files. Please move it to a utility file in `lib/src/__testutils__`

done 94307d7e404072e402735204b7137b776a18773f

File: lib/src/components/NxSearchTransferList/__tests__/NxSearchTransferList.test.tsx [hunk 5]
```diff
@@ -28,6 +29,10 @@ describe('NxSearchTransferList', function() {
...
+    mockTransferListLayout();
+  });
```
Comment (rpokorny) [inline lib/src/components/NxSearchTransferList/__tests__/NxSearchTransferList.test.tsx:34]:
```suggestion
  beforeEach(mockTransferListLayout);
```

**Review by rpokorny — APPROVED** [review]

File: lib/src/components/NxSearchTransferList/__tests__/NxSearchTransferList.test.tsx (hunk 5 above)
Comment (obarra-dev) [reply lib/src/components/NxSearchTransferList/__tests__/NxSearchTransferList.test.tsx]: > Likewise in the other files

done a348bc0051862123eac4fb8c4a467558c63c15b4

**Review by amoreno-sonatype — APPROVED** [review]
+1 LGTM

**Pull request merged**